fieldsolver/gridGlue.cpp

```diff
@@ -179,6 +189,244 @@ void feedMomentsIntoFsGrid(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
 
   MPI_Waitall(sendRequests.size(), sendRequests.data(), MPI_STATUSES_IGNORE);
 
+  // ---------------------------------------------------------------------------------
+  // Boxcar Smoothening
+  const int *mntDims= &momentsGrid.getLocalSize()[0];
+  std::array<Real,fsgrids::moments::N_MOMENTS> *Grid;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridL2;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR1;
+  std::array<Real,fsgrids::moments::N_MOMENTS> *GridR2;
```

Comment:
I would actually call these "cell" instead of "grid" since they are pointers to the data of a single grid cell each.
(Also: in the Vlasiator coding style, variable names start with lower case and datatype names with upper case!
See https://github.com/fmihpc/vlasiator/wiki/Coding-Style
)